python/lsst/ip/diffim/dcrModel.py

```diff
@@ -422,24 +389,29 @@ def buildMatchedTemplate(self, exposure=None, order=3,
         Raises
         ------
         ValueError
-            If neither ``exposure`` or all of ``visitInfo``, ``bbox``, and
-            ``wcs`` are set.
+            If neither ``exposure`` or ``visitInfo`` are set.
         """
         if self.effectiveWavelength is None or self.bandwidth is None:
             raise ValueError("'effectiveWavelength' and 'bandwidth' must be set for the DcrModel in order "
                              "to calculate DCR.")
         if exposure is not None:
             visitInfo = exposure.getInfo().getVisitInfo()
-            bbox = exposure.getBBox()
-            wcs = exposure.getInfo().getWcs()
-        elif visitInfo is None or bbox is None or wcs is None:
-            raise ValueError("Either exposure or visitInfo, bbox, and wcs must be set.")
-        dcrShift = calculateDcr(visitInfo, wcs, self.effectiveWavelength, self.bandwidth, len(self),
+        elif visitInfo is None:
+            raise ValueError("Either exposure or visitInfo must be set.")
+        if bbox is None:
+            bbox = self.bbox
+        dcrShift = calculateDcr(visitInfo, self.wcs, self.effectiveWavelength, self.bandwidth, len(self),
                                 splitSubfilters=splitSubfilters)
         templateImage = afwImage.ImageF(bbox)
-        refModel = self.getReferenceImage(bbox)
+        refModel = None
         for subfilter, dcr in enumerate(dcrShift):
+            if self[subfilter] is None:
+                # It is possible to load only a single DCR subfilter at a time.
+                self.log.debug("Skipping missing DCR model subfilter %d", subfilter)
+                continue
             if amplifyModel > 1:
+                if refModel is None:
+                    refModel = self.getReferenceImage(bbox)
```

Comment:
I'm not quite following the logic here. Before one would run self.getReferenceImage(bbox) once before the for loop. Now there is code to run it only the first time through the loop. What's the advantage? The only one I see is in case the first time through has a None subfilter, you would skip it. Am I missing something else?

Reply:
The advantage is that `getReferenceImage()` does not have to be run during image differencing. `amplifyModel` is an option that is only used while building the DcrCoadds. I'll add a comment to make it more clear.